v2/api/dbformysql/v1/user_types.go

```diff
@@ -166,14 +266,20 @@ type UserList struct {
 }
 
 type UserSpec struct {
-	//AzureName: The name of the resource in Azure. This is often the same as the name of the resource in Kubernetes but it
-	//doesn't have to be.
+	// AzureName: The name of the resource in Azure. This is often the same as the name of the resource in Kubernetes but it
+	// doesn't have to be.
+	// If not specified, the default is the name of the Kubernetes object.
+	// When creating a local user, this will be the name of the user created.
+	// When creating an AAD user, this must have a specific format depending on the type of AAD user being created.
+	// For managed identity: "my-managed-identity-name"
+	// For standard AAD user: "myuser@mydomain.onmicrosoft.com"
+	// For AAD group: "my-group"
```

Comment:
How does this syntax discriminate between managed-identity and AAD-group?

Reply:
As far as I can tell, it doesn't. 

I just confirmed that it is possible to create a UMI and a group with the same name. I've reached out to our MySQL contacts to ask this question to them, because [the public documentation](https://learn.microsoft.com/en-us/azure/mysql/flexible-server/how-to-azure-ad#create-azure-ad-groups-in-azure-database-for-mysql) doesn't really cover it.